### DISCUSSION TITLE
Add support for local large language models, (Ollama)

### DISCUSSION THREAD

**Discussion by iplayfast** [AliDehbansiahkarbon/ChatGPTWizard#57] [Ideas]
I have a network computer running linux, that I've set up an https://ollama.ai service on.

The steps to start the ollama service are:

installing & running ollama which is a llm server
installing litellm which is an openai emulator
running it.
```
litellm -m ollama/mistral --port 9000 --drop_params&

```
Which means I now have an openai compatible port at 192.168.0.15:9000 

Unfortunately when I tried using the ChatGPT wizard, it came back with an error "No answer, try again"
On the linux side the message said:
Docs: https://docs.litellm.ai/docs/proxy_server

Worker Initialized

INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:9000 (Press CTRL+C to quit)
192.168.0.196:50629 - "POST / HTTP/1.1" 405 Method Not Allowed

google show others with this problem.
https://github.com/openai/chatgpt-retrieval-plugin/issues/74

**Comment by AliDehbansiahkarbon** [comment]
Hi dear Chris,
**First of all, I appreciate you sharing the information with me, thank you.**

This is not an issue man 🫡, so I moved it into the discussions tab for now.
For your information, I have been exploring several open-source projects like this over the past three months.
I am waiting to find a powerful/suitable local project to integrate with this plugin. 
This one looks really cool, and I'm going to start working on it, 
So stay tuned🫵 something greater will...
